===================================================================
--- configs/common/Options.py
+++ configs/common/Options.py
@@ -222,6 +222,8 @@
         help="switch from timing to Detailed CPU after warmup period of <N>")
     parser.add_option("-p", "--prog-interval", type="str",
         help="CPU Progress Interval")
+    parser.add_option("--two-phase", action="store",
+                      help="switch from atomic to timing CPU after kernel boots")
 
     # Fastforwarding and simpoint related materials
     parser.add_option("-W", "--warmup-insts", action="store", type="int",
===================================================================
--- configs/common/Simulation.py
+++ configs/common/Simulation.py
@@ -78,7 +78,7 @@
         if options.restore_with_cpu != options.cpu_type:
             CPUClass = TmpClass
             TmpClass, test_mem_mode = getCPUClass(options.restore_with_cpu)
-    elif options.fast_forward:
+    elif options.fast_forward or options.two_phase:
         CPUClass = TmpClass
         TmpClass = AtomicSimpleCPU
         test_mem_mode = 'atomic'
@@ -639,6 +639,9 @@
             print "Switch at instruction count:%s" % \
                     str(testsys.cpu[0].max_insts_any_thread)
             exit_event = m5.simulate()
+        elif options.two_phase:
+            print "Switch after kernel boots"
+            exit_event = m5.simulate()
         else:
             print "Switch at curTick count:%s" % str(10000)
             exit_event = m5.simulate(10000)
@@ -688,7 +691,7 @@
         restoreSimpointCheckpoint()
 
     else:
-        if options.fast_forward:
+        if options.fast_forward or options.two_phase:
             m5.stats.reset()
         print "**** REAL SIMULATION ****"
 
